dart/test/http_client/sentry_http_client_test.dart

```diff
@@ -199,7 +180,48 @@ void main() {
       expect(mockHub.captureExceptionCalls.length, 0);
       verify(mockClient.close());
     });
+
+    test('Breadcrumb has correct duration', () async {
+      final mockHub = MockHub();
+
+      final mockClient = MockClient((request) async {
+        expect(request.url, requestUri);
+        await Future.delayed(Duration(seconds: 1));
+        return Response('', 200, reasonPhrase: 'OK');
+      });
+
+      final client = SentryHttpClient(client: mockClient, hub: mockHub);
+
+      final response = await client.get(requestUri);
+      expect(response.statusCode, 200);
+
+      expect(mockHub.addBreadcrumbCalls.length, 1);
+      final breadcrumb = mockHub.addBreadcrumbCalls.first.crumb;
+
+      var durationString = breadcrumb.data!['duration']! as String;
+      // we don't check for anything below a second
+      expect(durationString.startsWith('0:00:01'), true);
+    });
   });
 }
 
 class CloseableMockClient extends Mock implements BaseClient {}
+
+class Fixture {
+  Fixture() {
+    hub = MockHub();
+
+    final mockClient = MockClient((request) async {
+      expect(request.url, requestUri);
+      return Response('', response, reasonPhrase: reasonPhrase);
+    });
+
+    client = SentryHttpClient(client: mockClient, hub: hub);
```

Comment:
the tested class is always created under a `getSut()` method and not a global instance of `Fixture`, this would cause problems if tests are running in parallel.
you only need to create a fresh instance of `Fixture` in `setUp` if you are testing static variables, otherwise, only one instance of `Fixture` is needed, and a new `sut` is created for each test.